globals.nix

```diff
@@ -0,0 +1,98 @@
+{ IOHKaccessKeyId, CFaccessKeyId, EmurgoaccessKeyId
+, deployerIP
+, genesis                       ## Genesis file path or 'null'
+, topologyYaml                  ## The original stuff we're passing on to the nodes.
+, topologyFile ? ./topology.nix ## The iohk-ops post-processed thing.
+, systemStart
+, environment
+, ... }:
+
+with (import ./lib.nix);
+
+let topologySpec     = builtins.fromJSON (builtins.readFile topologyFile);
+    # WARNING: this sort order is key in a number of things:
+    #          - relay numbering
+    #          - DNS naming
+    topologySpecList = (builtins.sort (l: r: l.name < r.name)
+                                      (mapAttrsToList (k: v: { name = k; value = v;}) topologySpec))
+                       ++ [ explorerSpecElt reportServerSpecElt ];
+    # NOTE: the following definitions for explorerSpecElt and reportServerSpecElt
+    #       allow us to treat all cluster nodes in a uniform way.
+    #       It's as if they were defined in the topology yaml.
+    explorerSpecElt  = { name  = "explorer";
+                         value = { org      = defaultOrg;
+                                   region   = centralRegion;
+                                   zone     = centralZone;
+                                   type     = "other";
+                                   kademlia = false;
+                                   peers    = [];
+                                   address  = "explorer.cardano";
+                                   port     = 3000; }; };
+    reportServerSpecElt =
+                       { name  = "report-server";
+                         value = { org      = defaultOrg;
+                                   region   = centralRegion;
+                                   zone     = centralZone;
+                                   type     = "other";
+                                   kademlia = false;
+                                   peers    = [];
+                                   address  = "report-server.cardano";
+                                   port     = null; }; };
+
```

Comment:
№7

There is a tension between:
1. having simpler definitions for non-node machines
2. having a more uniform definition mechanism for all machines

I'm not sure this tension cannot be resolved at all -- we should be able to have both, with careful planning.

Reply:
This will be a result of DEVOPS-445